### DISCUSSION TITLE
REST API for admin functions

### DISCUSSION THREAD

**Discussion by pternyuk** [minio/minio#18280] [Q&A] [answered]
There was a discussion about rest api for mc admin functions few years ago (https://github.com/minio/minio/issues/7539)
I am looking for some way to interact with policy admin function (create, attach, detach and etc) via http-requests (or python scripts). I tried to debug traffic between mc utility and minio-server but faced difficulties with auth header. Maybe do you have plans to make documentation about auth header creation rules and admin endpoints?

Has something changed since that time?

**Comment by klauspost** [comment, marked as answer]
Policies are S3 API operations. You can already work with this in python: https://min.io/docs/minio/linux/developers/python/API.html#set_bucket_policy

We do not provide any Python API for admin functions, and have no current plans to do so. We provide a Go version: https://github.com/minio/madmin-go or you can interact with `mc` for scripts.
